simulator/memory/argv_loader/t/unit_tests.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Unit tests for argv_loader
+ * @author Rustem Yunusov <yunusov.rr@phystech.edu>
+ * Copyright 2019 MIPT-MIPS iLab project
+ */
+
+#include "../argv_loader.h"
+
+#include <catch.hpp>
+
+TEST_CASE( "ArgvLoader: load of argc and argv[]")
+{
+    const char* argv[4] = { "a", "b", "c"};
+    auto argv_loader = ArgvLoader( argv);
+    auto mem = FuncMemory::create_plain_memory( 20);
+    CHECK( argv_loader.load_argv_to( mem, 0) == 42);
+    CHECK( ( mem->read<int,    Endian::little>( 0)) == 3);
+    CHECK( ( mem->read<uint64, Endian::little>( 4))  == uint64( argv[0]));
+    CHECK( ( mem->read<uint64, Endian::little>( 12)) == uint64( argv[1]));
+    CHECK( ( mem->read<uint64, Endian::little>( 20)) == uint64( argv[2]));
+    CHECK( ( mem->read<uint64, Endian::little>( 28)) == 0);
+    CHECK( ( mem->read<uint16, Endian::little>( 36)) == 'a');
+    CHECK( ( mem->read<uint16, Endian::little>( 38)) == 'b');
+    CHECK( ( mem->read<uint16, Endian::little>( 40)) == 'c');
+}
+
+TEST_CASE( "ArgvLoader: load of argc, argv[] and envp[]")
+{
+    const char* argv[4] = { "a", "b", "c"};
+    const char* envp[3] = { "d", "e"};
+    auto argv_loader = ArgvLoader( argv, envp);
+    auto mem = FuncMemory::create_plain_memory( 20);
+    CHECK( argv_loader.load_argv_to( mem, 0) == 70);
+    CHECK( ( mem->read<int,    Endian::little>( 0)) == 3);
+    CHECK( ( mem->read<uint64, Endian::little>( 4))  == uint64( argv[0]));
+    CHECK( ( mem->read<uint64, Endian::little>( 12)) == uint64( argv[1]));
+    CHECK( ( mem->read<uint64, Endian::little>( 20)) == uint64( argv[2]));
```

Comment:
You are wrong here!
You're writing pointer to the **host** memory to **guest** memory.